### PR TITLE
Calendar: Updating styles to match new OWA calendar

### PR DESCRIPTION
**Pull request by kellygorr**
#### Pull request checklist

[0] Addresses an existing issue: Fixes #0000
[X] Include a change request file using `$ npm run change`

#### Description of changes

Updating Calendar component styles to match new OWA calendar styles for the Calendar team.

Inline calendar- day highlighted BEFORE/AFTER
inlinebefore
inlineafter

Inline calendar- week highlighted BEFORE/AFTER
inline3before
inline3after

Inline calendar- month highlighted BEFORE/AFTER
inline2before
inline2after

Month-only BEFORE/AFTER
monthonlybefore
monthonly

"Click for calendar" BEFORE/AFTER
fullbefore
fullafter

### CODE REVIEW

**Comment by micahgodbolt** [comment]
BTW, awesome PR with pictures! Fix up that conflict and once @johannao76 or someone else on the OWA team has a chance to review, we can merge this in.

**Comment by dzearing** [comment]
We should not round button corners in one component but not others. sent a chat message to designers; let's get an ok to be consistent everywhere before we introduce inconsistency please.

File: packages/office-ui-fabric-react/src/components/Calendar/CalendarDay.tsx [hunk 1]
```diff
@@ -204,6 +227,162 @@ export class CalendarDay extends BaseComponent<ICalendarDayProps, ICalendarDaySt
...
+
+    // Highlighted days in first week of month
+    let firstWeek: any[] = [];
+    weeks[0].map((day: any, index: number) => {
+      if (day.isInMonth) { firstWeek.push(index); }
+    });
+
+    // Highlighted days in last week of month
+    let lastWeek: any[] = [];
+    weeks[numberOfWeeks - 1].map((day: any, index: number) => {
```
Comment (johannao76) [inline packages/office-ui-fabric-react/src/components/Calendar/CalendarDay.tsx]: Would this be an easier way to determine where to put rounded corners? It avoids all the looping:
- first day of range: top/left style
- first day in current view + 7 (use addDays function?): top/right style
- first day in current view + 7 + 1 (if delta with first day in range < 7 days): top/left style
- last day in range: bottom/right style
- last day in current view - 7 (use addDays function with negative value?: bottom/right style 
- last day in current view - 7 - 1 (if delta with last day in range < 7 days): bottom/right style

File: packages/office-ui-fabric-react/src/components/Calendar/CalendarDay.tsx [hunk 2]
```diff
@@ -204,6 +227,162 @@ export class CalendarDay extends BaseComponent<ICalendarDayProps, ICalendarDaySt
...
+    let cornerStyle = '0px';
+    if (weekCorners) {
+      weekCorners.forEach((corner: any) => {
+        if (corner.index === dayIndex && corner.week === weekIndex) {
+          cornerStyle = corner.styles;
+        }
+      });
+    }
+
+    return cornerStyle;
```
Comment (johannao76) [inline packages/office-ui-fabric-react/src/components/Calendar/CalendarDay.tsx]: This function is actually returning a style so the function should probably be renamed to getSelectedCornerStyle(). Also, instead of looping, it might be more efficient to use a dictionary for your weekCorners, and you can key off of a combination of day and week, i.e., key: "week_day".

File: packages/office-ui-fabric-react/src/components/Calendar/CalendarMonth.tsx [hunk 3]
```diff
@@ -101,7 +101,8 @@ export class CalendarMonth extends BaseComponent<ICalendarMonthProps, {}> {
                   css('ms-DatePicker-monthOption',
                     styles.monthOption,
                     {
-                      ['ms-DatePicker-day--today ' + styles.monthIsCurrentMonth]: highlightCurrentMonth && this.isCurrentMonth(index, navigatedDate.getFullYear(), today!)
+                      ['ms-DatePicker-day--today ' + styles.monthIsCurrentMonth]: highlightCurrentMonth && this.isCurrentMonth(index, navigatedDate.getFullYear(), today!),
+                      ['ms-DatePicker-day--highlighted ' + styles.monthIsHighlighted]: highlightCurrentMonth && (navigatedDate.getMonth() === index)
```
Comment (johannao76) [inline packages/office-ui-fabric-react/src/components/Calendar/CalendarMonth.tsx]: Isn't line 105 intending to do the same check as line 104? what is this monthIsHighlighted for?

File: packages/office-ui-fabric-react/src/components/Calendar/CalendarPage.tsx [hunk 4]
```diff
@@ -26,10 +25,10 @@ export class CalendarPage extends React.Component<IComponentDemoPageProps, any>
             </ExampleCard>
             <ExampleCard title='Inline Calendar with overlayed month picker when header is clicked' code={ CalendarInlineExampleCode }>
               <CalendarInlineExample showMonthPickerAsOverlay={ true } highlightCurrentMonth={ true } dateRangeType={ DateRangeType.Day } autoNavigateOnSelection={ false }
-                showGoToToday={ true } />
+                showGoToToday={ false } />
             </ExampleCard>
             <ExampleCard title='Inline Calendar with month picker' code={ CalendarInlineExampleCode }>
-              <CalendarInlineExample dateRangeType={ DateRangeType.Day } autoNavigateOnSelection={ false }
+              <CalendarInlineExample dateRangeType={ DateRangeType.Day } autoNavigateOnSelection={ false } highlightCurrentMonth={ true }
```
Comment (johannao76) [inline packages/office-ui-fabric-react/src/components/Calendar/CalendarPage.tsx]: I suggest you add a test for when the first day of week is something other than the default and you are using a custom date range like month, to make sure the corners are correct

File: packages/office-ui-fabric-react/src/components/Calendar/Calendar.scss [hunk 5]
```diff
@@ -11,6 +11,7 @@ $Calendar-dayPicker-margin: 10px;
 
 .root {
   @include ms-normalize;
+  letter-spacing: -0.1px;
```
Comment (johannao76) [inline packages/office-ui-fabric-react/src/components/Calendar/Calendar.scss]: Why do we need this? Seems a bit arbitrary.

File: packages/office-ui-fabric-react/src/components/Calendar/Calendar.scss [hunk 6]
```diff
@@ -119,6 +122,7 @@ $Calendar-dayPicker-margin: 10px;
   line-height: $Calendar-day;
   @include ms-font-m-plus;
   color: $ms-color-neutralPrimary;
+  border-radius: 2px;
```
Comment (johannao76) [inline packages/office-ui-fabric-react/src/components/Calendar/Calendar.scss]: Please apply all my feedback first with rounded corners enabled to make sure the calculations work, but before committing these changes, let's send out a final iteration of the PR with the rounded corners commented out.

File: packages/office-ui-fabric-react/src/components/Calendar/Calendar.scss [hunk 7]
```diff
@@ -162,42 +165,61 @@ $Calendar-dayPicker-margin: 10px;
...
-  display: none;
+//Background for highlighted day
+.dayBackground{
+  background: $ms-color-neutralLight;
+  border-radius: 2px;
 }
 
+//Background for highlighted week
+.weekBackground{
+    background: $ms-color-neutralLight;
```
Comment (johannao76) [inline packages/office-ui-fabric-react/src/components/Calendar/Calendar.scss]: These are the same for day/week/month. I would consolidate these

File: packages/office-ui-fabric-react/src/components/Calendar/Calendar.scss [hunk 8]
```diff
@@ -337,43 +354,34 @@ $Calendar-dayPicker-margin: 10px;
...
 
-    // Update the spacing and text for the day and month pickers
+  // Update the spacing and text for the day and month pickers
 
-    .header {
-      height: $Calendar-day;
-      line-height: $Calendar-day;
-    }
+  .dayPicker,.monthPicker {
+    min-height: ( $Calendar-day * 8 ) - $Calendar-dayPicker-margin;
```
Comment (johannao76) [inline packages/office-ui-fabric-react/src/components/Calendar/Calendar.scss]: Why do we need this math in the styles? More efficient to not have these computations

File: packages/office-ui-fabric-react/src/components/Calendar/Calendar.scss [hunk 9]
```diff
@@ -409,43 +411,36 @@ $Calendar-dayPicker-margin: 10px;
...
       margin: -$Calendar-dayPicker-margin 0;
       padding: $Calendar-dayPicker-margin 0;
     }
+
     // Contains the calendar view for picking a day.
     .dayPicker {
       @include ms-borderBox;
       @include border-right(1px, solid, $ms-color-neutralLight);
-      width: 220px;
+      width: 212px;
```
Comment (johannao76) [inline packages/office-ui-fabric-react/src/components/Calendar/Calendar.scss]: Why are we changing this? We had some pushback when I made the initial change to reduce the width here. Is this change strictly necessary for these changes?

File: packages/office-ui-fabric-react/src/components/Calendar/Calendar.scss [hunk 10]
```diff
@@ -409,43 +411,36 @@ $Calendar-dayPicker-margin: 10px;
...
-    }
-    // Size and position of the month and year labels.
-    .month {
-      @include margin-left(12px);
-    }
 
     .optionGrid {
-      width: 200px;
-      @include margin(10px, 0, 0, 0);
+      height: ( ($Calendar-month + 10) * 3 );
```
Comment (johannao76) [inline packages/office-ui-fabric-react/src/components/Calendar/Calendar.scss]: avoid computations

File: packages/office-ui-fabric-react/src/components/Calendar/Calendar.scss [hunk 11]
```diff
@@ -649,10 +577,69 @@ $Calendar-dayPicker-margin: 10px;
...
+  }
+}
+
+.monthOption:active{
   background-color: $ms-color-themeLight;
 }
+
+
+//Rounded corners
+.roundedCornersTopLeft{
```
Comment (johannao76) [inline packages/office-ui-fabric-react/src/components/Calendar/Calendar.scss]: Rename these classes to something more functional, i.e., selectedTopLeftCornerDate

File: packages/office-ui-fabric-react/src/components/Calendar/Calendar.test.tsx [hunk 12]
```diff
@@ -99,7 +99,7 @@ describe('Calendar', () => {
 
     it('Verify day picker header', () => {
       let today = new Date();
-      let monthName = dayPickerStrings.months[today.getMonth()];
+      let monthName = dayPickerStrings.shortMonths[today.getMonth()];
```
Comment (kellygorr) [inline packages/office-ui-fabric-react/src/components/Calendar/Calendar.test.tsx]: I thought this was more of a style change, the months need to be short or the layout will break with longer months.

File: packages/office-ui-fabric-react/src/components/Calendar/Calendar.tsx [hunk 13]
```diff
@@ -120,9 +119,8 @@ export class Calendar extends BaseComponent<ICalendarProps, ICalendarState> impl
           styles.picker,
           styles.pickerIsOpened,
           styles.pickerIsFocused,
-          showSmallMonthPicker && ('is-monthPickerVisible ' + styles.pickerIsMonthPickerVisible),
-          isMonthPickerVisible && !isDayPickerVisible && !showMonthPickerAsOverlay && ('is-onlymonthPickerVisible ' + styles.pickerOnlyMonthPickerVisible),
-          showMonthPickerAsOverlay && styles.overlayedCalendars,
+          isMonthPickerVisible && ('ms-DatePicker-monthPickerVisible ' + styles.monthPickerVisible),
+          isMonthPickerVisible && isDayPickerVisible && ('ms-DatePicker-calendarsInline ' + styles.calendarsInline),
```
Comment (kellygorr) [inline packages/office-ui-fabric-react/src/components/Calendar/Calendar.tsx]: Yes, they are not needed.  The month-only calendar was being styled with an additional set of styles (pickerOnlyMonthPickerVisible) that made it 'oversized', since we don't want the month-only calendar to be that large, I was able to remove those styles and simplify how all the styles were applied.

File: packages/office-ui-fabric-react/src/components/Calendar/CalendarDay.tsx [hunk 14]
```diff
@@ -86,20 +86,31 @@ export class CalendarDay extends BaseComponent<ICalendarDayProps, ICalendarDaySt
...
+        weekCorners = this.findWeekCorners(weeks);
+      } else if (nextMonth.getMonth() === navigatedDate.getMonth() && nextMonth.getFullYear() === navigatedDate.getFullYear()) {
+        // navigatedDate is next month
+        weekCorners = this.findUnFocusedWeekCorners(weeks);
+      } else if (previousMonth.getMonth() === navigatedDate.getMonth() && previousMonth.getFullYear() === previousMonth.getFullYear()) {
+        // navigatedDate is previous month
+        weekCorners = this.findUnFocusedWeekCorners(weeks);
+      }
+    }
+
```
Comment (kellygorr) [inline packages/office-ui-fabric-react/src/components/Calendar/CalendarDay.tsx]: Yes!  Removing this!

File: packages/office-ui-fabric-react/src/components/Calendar/CalendarMonth.tsx (hunk 3 above)
Comment (kellygorr) [inline packages/office-ui-fabric-react/src/components/Calendar/CalendarMonth.tsx]: When the day and month picker are inline with each other there is a currentMonth and a highlighted month if the select a different month.

File: packages/office-ui-fabric-react/src/components/Calendar/Calendar.scss (hunk 5 above)
Comment (kellygorr) [inline packages/office-ui-fabric-react/src/components/Calendar/Calendar.scss]: It was in the zeplin styles for all the text, so that is why I added it.  But I will remove!

File: packages/office-ui-fabric-react/src/components/Calendar/Calendar.scss (hunk 11 above)
Comment (kellygorr) [inline packages/office-ui-fabric-react/src/components/Calendar/Calendar.scss]: Yes, will do!

File: packages/office-ui-fabric-react/src/components/Calendar/Calendar.scss (hunk 10 above)
Comment (kellygorr) [inline packages/office-ui-fabric-react/src/components/Calendar/Calendar.scss]: They are using computations for the day picker
.dayPicker,.monthPicker {
    min-height: ( $Calendar-day * 8 ) - $Calendar-dayPicker-margin;
  }

So that is why I was using a similar function to set the .optionGrid height.

File: packages/office-ui-fabric-react/src/components/Calendar/Calendar.scss (hunk 9 above)
Comment (kellygorr) [inline packages/office-ui-fabric-react/src/components/Calendar/Calendar.scss]: The daypicker looks to be 212px in the comps

File: packages/office-ui-fabric-react/src/components/Calendar/Calendar.scss (hunk 7 above)
Comment (kellygorr) [inline packages/office-ui-fabric-react/src/components/Calendar/Calendar.scss]: Yes!

File: packages/office-ui-fabric-react/src/components/Calendar/CalendarDay.tsx (hunk 1 above)
Comment (kellygorr) [inline packages/office-ui-fabric-react/src/components/Calendar/CalendarDay.tsx]: This looks like it will work!  I will give it a try!

File: packages/office-ui-fabric-react/src/components/Calendar/CalendarDay.tsx (hunk 2 above)
Comment (kellygorr) [reply packages/office-ui-fabric-react/src/components/Calendar/CalendarDay.tsx]: Yes!  Great idea!

File: packages/office-ui-fabric-react/src/components/Calendar/CalendarPage.tsx (hunk 4 above)
Comment (kellygorr) [reply packages/office-ui-fabric-react/src/components/Calendar/CalendarPage.tsx]: Hi!  Not sure what you mean, the function that looks for the month corners only runs when the DateRanteType === 2 (month), it wouldn't run with DateRangeType.Day.  Lets discuss Monday!

File: packages/office-ui-fabric-react/src/components/Calendar/Calendar.test.tsx (hunk 12 above)
[the remark this answers is not in the document]
Comment (johannao76) [reply packages/office-ui-fabric-react/src/components/Calendar/Calendar.test.tsx]: If this is for the main header, then this needs to be the full month and not short month, per JSMVVM implementation.

File: packages/office-ui-fabric-react/src/components/Calendar/Calendar.tsx (hunk 13 above)
[the remark this answers is not in the document]
Comment (johannao76) [reply packages/office-ui-fabric-react/src/components/Calendar/Calendar.tsx]: Cool

File: packages/office-ui-fabric-react/src/components/Calendar/CalendarDay.tsx [hunk 15]
```diff
@@ -204,6 +219,76 @@ export class CalendarDay extends BaseComponent<ICalendarDayProps, ICalendarDaySt
     }
   }
 
+  private findWeekCorners(weeks: any) {
+    let weekCorners: any = {};
+    let numberOfWeeks = weeks.length;
+    let { firstDayOfWeek, navigatedDate, dateRangeType } = this.props;
+
+    // Get all days in current month
+    let dateRange = getDateRangeArray(navigatedDate, dateRangeType, firstDayOfWeek);
```
Comment (johannao76) [inline packages/office-ui-fabric-react/src/components/Calendar/CalendarDay.tsx]: I don;t think you need to get all days in the month just to find the dates you want. Can't you use the addDays function?

File: packages/office-ui-fabric-react/src/components/Calendar/CalendarDay.tsx [hunk 16]
```diff
@@ -204,6 +219,76 @@ export class CalendarDay extends BaseComponent<ICalendarDayProps, ICalendarDaySt
...
+
+    //Check to see if second to the last day (of the first week) is within the current month
+    if (weeks[0][DAYS_IN_WEEK - 2].originalDate.getMonth() !== navigatedDate.getMonth()) {
```
Comment (johannao76) [inline packages/office-ui-fabric-react/src/components/Calendar/CalendarDay.tsx]: The weeks already have days and for each day, you already know if the day is in the month, check out the isInMonth property

File: packages/office-ui-fabric-react/src/components/Calendar/CalendarDay.tsx [hunk 17]
```diff
@@ -86,20 +86,23 @@ export class CalendarDay extends BaseComponent<ICalendarDayProps, ICalendarDaySt
...
-    let { firstDayOfWeek, strings, navigatedDate, navigationIcons } = this.props;
+    let { firstDayOfWeek, strings, navigatedDate, navigationIcons, dateRangeType, selectedDate } = this.props;
     let dayPickerId = getId('DatePickerDay-dayPicker');
     let monthAndYearId = getId('DatePickerDay-monthAndYear');
     let leftNavigationIcon = navigationIcons.leftNavigation;
     let rightNavigationIcon = navigationIcons.rightNavigation;
 
+    // When the month is highlighted get the corner dates so that styles can be added to them
+    let weekCorners: any[] = [];
+    if (dateRangeType === 2) {
```
Comment (johannao76) [inline packages/office-ui-fabric-react/src/components/Calendar/CalendarDay.tsx]: Avoid hardcoding the enum values in this way

File: packages/office-ui-fabric-react/src/components/Calendar/CalendarDay.tsx [hunk 18]
```diff
@@ -86,20 +86,23 @@ export class CalendarDay extends BaseComponent<ICalendarDayProps, ICalendarDaySt
...
     let { activeDescendantId, weeks } = this.state;
-    let { firstDayOfWeek, strings, navigatedDate, navigationIcons } = this.props;
+    let { firstDayOfWeek, strings, navigatedDate, navigationIcons, dateRangeType, selectedDate } = this.props;
     let dayPickerId = getId('DatePickerDay-dayPicker');
     let monthAndYearId = getId('DatePickerDay-monthAndYear');
     let leftNavigationIcon = navigationIcons.leftNavigation;
     let rightNavigationIcon = navigationIcons.rightNavigation;
 
+    // When the month is highlighted get the corner dates so that styles can be added to them
+    let weekCorners: any[] = [];
```
Comment (johannao76) [inline packages/office-ui-fabric-react/src/components/Calendar/CalendarDay.tsx]: Use an interface type instead of any if you can

File: packages/office-ui-fabric-react/src/components/Calendar/CalendarDay.tsx [hunk 19]
```diff
@@ -86,20 +86,23 @@ export class CalendarDay extends BaseComponent<ICalendarDayProps, ICalendarDaySt
...
+    if (dateRangeType === 2) {
+      if (selectedDate.getMonth() === navigatedDate.getMonth() && selectedDate.getFullYear() === navigatedDate.getFullYear()) {
```
Comment (johannao76) [inline packages/office-ui-fabric-react/src/components/Calendar/CalendarDay.tsx]: Combine with the if statement in the previous line

File: packages/office-ui-fabric-react/src/components/Calendar/CalendarDay.tsx [hunk 20]
```diff
@@ -204,6 +219,76 @@ export class CalendarDay extends BaseComponent<ICalendarDayProps, ICalendarDaySt
     }
   }
 
+  private findWeekCorners(weeks: any) {
```
Comment (johannao76) [inline packages/office-ui-fabric-react/src/components/Calendar/CalendarDay.tsx]: Avoid passing any, use IDayInfo[][]

File: packages/office-ui-fabric-react/src/components/Calendar/CalendarDay.tsx [hunk 21]
```diff
@@ -204,6 +219,76 @@ export class CalendarDay extends BaseComponent<ICalendarDayProps, ICalendarDaySt
...
+    if (weeks[0][DAYS_IN_WEEK - 2].originalDate.getMonth() !== navigatedDate.getMonth()) {
+      //There is only one highlighted day in first week, add styles to this square
+      let weekIndex = 0;
+      let dayIndex = DAYS_IN_WEEK - 1;
+      weekCorners[weekIndex + '_' + dayIndex] = 'ms-DatePicker-singleTopDate ' + styles.singleTopDate;
+    }
+    else {
+      //Add styles to first and last highlighted squares in the first week
+      let weekIndex = 0;
+      let dayIndex = dateRange[0].getDay();
```
Comment (johannao76) [inline packages/office-ui-fabric-react/src/components/Calendar/CalendarDay.tsx]: This is not going to work with custom first day of the week, please be sure to test this scenario carefully.

File: packages/office-ui-fabric-react/src/components/Calendar/CalendarDay.tsx [hunk 22]
```diff
@@ -204,6 +219,76 @@ export class CalendarDay extends BaseComponent<ICalendarDayProps, ICalendarDaySt
...
+      weekCorners[weekIndex + '_' + dayIndex] = 'ms-DatePicker-topLeftCornerDate ' + styles.topLeftCornerDate;
+
+      let weekIndex2 = 0;
```
Comment (johannao76) [inline packages/office-ui-fabric-react/src/components/Calendar/CalendarDay.tsx]: You dont need this var, this is the same as weekIndex1

File: packages/office-ui-fabric-react/src/components/Calendar/CalendarDay.tsx [hunk 23]
```diff
@@ -204,6 +219,76 @@ export class CalendarDay extends BaseComponent<ICalendarDayProps, ICalendarDaySt
...
+      let dayIndex2 = DAYS_IN_WEEK - 1;
+      weekCorners[weekIndex2 + '_' + dayIndex2] = 'ms-DatePicker-topRightCornerDate ' + styles.topRightCornerDate;
+    }
+
+    //check the first week to see if the first day of the week is NOT within the current month
+    if (weeks[0][0].originalDate.getMonth() !== navigatedDate.getMonth()) {
```
Comment (johannao76) [inline packages/office-ui-fabric-react/src/components/Calendar/CalendarDay.tsx]: But you already knew from the checks above which was the index of the first and last highlighted days of the first week, so you could have used that info, no need to check this again.

File: packages/office-ui-fabric-react/src/components/Calendar/CalendarDay.tsx [hunk 24]
```diff
@@ -204,6 +219,76 @@ export class CalendarDay extends BaseComponent<ICalendarDayProps, ICalendarDaySt
...
+
+    return weekCorners;
+  }
+
+  private getHighlightedCornerStyle(weekCorners: any, dayIndex: number, weekIndex: number) {
+    let cornerStyle = weekCorners[weekIndex + '_' + dayIndex] ? weekCorners[weekIndex + '_' + dayIndex] : ''
+
+    return cornerStyle;
+  }
+
```
Comment (johannao76) [inline packages/office-ui-fabric-react/src/components/Calendar/CalendarDay.tsx]: Sent you feedback offline on a possibly better way to cleanup this code

File: packages/office-ui-fabric-react/src/components/Calendar/CalendarPage.tsx (hunk 4 above)
Comment (kellygorr) [reply packages/office-ui-fabric-react/src/components/Calendar/CalendarPage.tsx]: This is tested and working with the updated from 8-30-17

File: packages/office-ui-fabric-react/src/components/Calendar/CalendarDay.tsx [hunk 25]
```diff
@@ -204,6 +221,80 @@ export class CalendarDay extends BaseComponent<ICalendarDayProps, ICalendarDaySt
...
+      weekCornersStyled[lastWeekIndex + '_' + indexesLastWeek[0]] = 'ms-DatePicker-bottomRightCornerDate ' + styles.bottomRightCornerDate;
+      weekCornersStyled[lastWeekIndex + '_' + indexesLastWeek[1]] = 'ms-DatePicker-bottomLeftCornerDate ' + styles.bottomLeftCornerDate;
+    }
+
+    // check if second-to-last week needs corner styles
+    if (!weeks[lastWeekIndex][DAYS_IN_WEEK - 1].isInMonth) {
+      weekCornersStyled[(lastWeekIndex - 1) + '_' + (DAYS_IN_WEEK - 1)] = 'ms-DatePicker-bottomRightCornerDate ' + styles.bottomRightCornerDate;
+    }
+
+    return weekCornersStyled;
```
Comment (johannao76) [inline packages/office-ui-fabric-react/src/components/Calendar/CalendarDay.tsx]: There are a bunch of undefined values in the output of this function, it looks like none of the styles referenced are defined...

File: packages/office-ui-fabric-react/src/components/Calendar/CalendarDay.tsx [hunk 26]
```diff
@@ -204,6 +221,80 @@ export class CalendarDay extends BaseComponent<ICalendarDayProps, ICalendarDaySt
...
+        let lastDayIndex = lastDayofMonth.getDay() - firstDayOfWeek;
+        lastDayIndex = lastDayIndex < 0 ? lastDayIndex + DAYS_IN_WEEK : lastDayIndex;
+        cornerIndexes.push(lastDayIndex);
+      }
+
+      cornerIndexes.push(0);
+    }
+
+    return cornerIndexes;
+  }
```
Comment (johannao76) [inline packages/office-ui-fabric-react/src/components/Calendar/CalendarDay.tsx]: This function can be vastly simplified to something like this:
  private _findCornerIndexes(week: IDayInfo[]) {
    let cornerIndexes = [];
    for (let i = 0, length = week.length; i < length; i++) {
      let day = week[i];
      if (day.isInMonth) {
        cornerIndexes.push(i);
      }
    }
    if (cornerIndexes.length > 2) {
      cornerIndexes.splice(1, cornerIndexes.length - 2);
    }
    return cornerIndexes;
  }

File: packages/office-ui-fabric-react/src/components/Calendar/CalendarDay.tsx (hunk 25 above)
Comment (johannao76) [reply packages/office-ui-fabric-react/src/components/Calendar/CalendarDay.tsx]: I also sent you a slightly updated version of this code that is a bit cleaner
